specs/CDPSessionIdSupport.md

```diff
@@ -0,0 +1,316 @@
+sessionId support for DevToolsProtocol method call and event
+===
+
+# Background
+A web page can have multiple DevToolsProtocol targets. Besides the default target for the top page, there are separate targets for iframes from different origin and web workers.
+
+The underlying DevToolsProtocol supports interaction with other targets by attaching to them and then using sessionId of the attachment in DevToolsProtocol method call message.
+The received DevToolsProtocol event messages also have sessionId field to indicate which target the event comes from.
+See [DevToolsProtocol Target domain](https://chromedevtools.github.io/devtools-protocol/tot/Target/) for more details.
+
+However, the current WebView2 DevToolsProtocol APIs like [CallDevToolsProtocolMethod](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2#calldevtoolsprotocolmethod)
+and [DevToolsProtocolEventReceivedEvent](https://docs.microsoft.com/en-us/microsoft-edge/webview2/reference/win32/icorewebview2devtoolsprotocoleventreceivedeventargs)
+doesn't support sessionId.
+
+To support interaction with different parts of the page, we are adding support for specifying sessionId for DevToolsProtocol method call API
+and retrieving sessionId for received DevToolsProtocol events.
+  
+# Conceptual pages (How To)
+
+To use the sessionId support, you must attach to targets with with `flatten` set as `true` when calling `Target.attachToTarget` or `Target.setAutoAttach`.
+
+You can listen to `Target.attachedToTarget` and `Target.detachedFromTarget` events to manage the sessionId for targets, and listen to `Target.targetInfoChanged` event to update target info like url of a target.
+
+There is also some nuance for DevToolsProtocol's target management. If you are interested in only top page and iframes from different origins on the page, it will be simple and straight forward. All related methods and events like `Target.getTargets`, `Target.attachToTarget`, and `Target.targetCreated` event work as expected.
+
+However, dedicated web workers are not returned from `'Target.getTargets'`, and you have to call DevToolsProtocol method `Target.setAutoAttach` to be able to attach to them.
+
+And shared worker is separate from any page or iframe target, and therefore will not be auto attached. You have to call `Target.attachToTarget` to attach to them. The shared workers can be enumerated with `Target.getTargets`. They are also discoverable, that is, you can call `Target.setDiscoverTargets` to receive `Target.targetCreated` event when a shared worker is created.
+
```

Comment:
So there's two ways of finding the targets and neither covers all 3 scenarios? It might be helpful to have a table summarizing it. Is the following correct?

|      | Web pages | Dedicated web workers | Shared workers |
| --- | --- | --- | --- |
| Target.getTargets, Target.setDiscoverTargets, Target.targetCreated, Target.attachToTarget | ✔ | | ✔ |
| Target.setAutoAttach, Target.attachedToTarget | ✔ | ✔ |  |

Reply:
Added the table